src/hooks/useAuthorityLinking/useAuthorityLinking.js

```diff
@@ -149,13 +184,15 @@ const useAuthorityLinking = ({ tenantId, marcType, action } = {}) => {
       newZeroSubfield = authorityRecord.naturalId;
     }
 
-    bibSubfields.$0 = [newZeroSubfield];
+    bibSubfields.removeByCode('$0');
+    bibSubfields.append('$0', newZeroSubfield);
 
     const updatedBibSubfields = copySubfieldsFromAuthority(bibSubfields, linkedAuthorityField, bibField.tag);
 
-    updatedBibSubfields.$9 = [authorityRecord.id];
+    updatedBibSubfields.removeByCode('$9');
+    updatedBibSubfields.append('$9', authorityRecord.id);
     bibField.prevContent = bibField.content;
-    bibField.content = joinSubfields(updatedBibSubfields);
+    bibField.content = updatedBibSubfields.join();

```

Comment:
I wonder if it would be better to avoid pretending to be array methods. This implementation looks cool, but is confusing because it is not an array, but uses `join`, which is actually `reduce`.